server/re/src/irods_re_serialization.cpp

```diff
@@ -1199,6 +1199,77 @@ namespace irods::re_serialization
         }
     } // serialize_vector_of_strings_ptr
 
+    static irods::error serialize_execMyRuleInp_ptr(boost::any _p, serialized_parameter_t& _out)
+    {
+        try {
+            log_re::trace(__func__);
+            const auto* casted_ptr = boost::any_cast<execMyRuleInp_t*>(_p);
+            if(!casted_ptr) {
+                _out["execMyRuleInp_ptr"] = "nullptr";
+                return SUCCESS();
+            }
+            _out["myRule"] = casted_ptr->myRule;
+            _out["addr_hostAddr"] = casted_ptr->addr.hostAddr;
+            _out["addr_zoneName"] = casted_ptr->addr.zoneName;
+            _out["addr_portNum"] = boost::lexical_cast<std::string>(casted_ptr->addr.portNum);
+            _out["addr_dummyInt"] = boost::lexical_cast<std::string>(casted_ptr->addr.dummyInt);
+            serialize_keyValPair(casted_ptr->condInput, _out);
+            _out["outParamDesc"] = casted_ptr->outParamDesc;
+
+            const auto* msparamary_ptr = casted_ptr->inpParamArray;
+            if(!msparamary_ptr) {
+                _out["inpParamArray"] = "nullptr";
+                return SUCCESS();
+            }
+            _out["inpParamArray_len"] = boost::lexical_cast<std::string>(msparamary_ptr->len);
+            _out["inpParamArray_oprType"] = boost::lexical_cast<std::string>(msparamary_ptr->oprType);
+            for (int i = 0; i < msparamary_ptr->len; i++) {
+                std::string prefix = "inpParamArray_" + std::to_string(i) + "_";
+                _out[prefix + "label"] = msparamary_ptr->msParam[i]->label;
+                _out[prefix + "type"] = msparamary_ptr->msParam[i]->type;
+                const auto* msparam_ptr = msparamary_ptr->msParam[i];
+
+                if (msparam_ptr == NULL || msparam_ptr->inOutStruct == NULL) {
+                    _out[prefix + "inOutStruct"] = "null";
+                }
+                else if (strcmp(msparam_ptr->type, STR_MS_T) != 0) {
+                    _out[prefix + "inOutStruct"] = "non_string_value";
+                }
+                else {
+                    _out[prefix + "inOutStruct"] = static_cast<char*>(msparam_ptr->inOutStruct);
+                }
```

Comment:
Probably should add support for a few other types as well.

See the following.
https://github.com/irods/irods/blob/669a726968dcdd7c4dab7d52761095119511c182/server/core/src/json_serialization.cpp#L143-L157

Reply:
Added these, but I don't know how to verify if it works. `irule` seems to want to pass in all arguments as strings. Any way around that?

Reply:
I don't think there is.  I think another client could potentially submit other types - but I think testing for the strings is probably the 95%+ use case.

Reply:
Right. Covering the string case is good enough. The extra types are just me wanting to cover potential cases.